### PR TITLE
feat(lib): add docstrings to artifactory/azurerm/consul backend resources

### PR DESCRIPTION
**Pull request by aayushharwani-aidash**
Have added doc-string to artifactory,azurerm and consul backend resources

Working on issue - [https://github.com/hashicorp/terraform-cdk/issues/1268](https://github.com/hashicorp/terraform-cdk/issues/1268)

### CODE REVIEW

**Comment by aayushharwani-aidash** [comment]
@ansgarm

File: packages/cdktf/lib/backends/azurerm-backend.ts [hunk 1]
```diff
@@ -36,21 +36,91 @@ export class DataTerraformRemoteStateAzurerm extends TerraformRemoteState {
...
+ * 
+ * This backend supports state locking and consistency checking
+ * with Azure Blob Storage native capabilities.
+ * 
+ * Note: By default the Azure Backend uses ADAL for authentication which is deprecated 
+ * in favour of MSAL - MSAL can be used by setting use_microsoft_graph to true. 
+ * The default for this will change in Terraform 1.2, 
+ * so that MSAL authentication is used by default.
+ * 
+ * https://www.terraform.io/language/settings/backends/azurerm
```
Comment (ansgarm) [inline packages/cdktf/lib/backends/azurerm-backend.ts:51]:
```suggestion
 * Read more about this backend in the Terraform docs:
 * https://www.terraform.io/language/settings/backends/azurerm
```

File: packages/cdktf/lib/backends/consul-backend.ts [hunk 2]
```diff
@@ -36,18 +36,64 @@ export class DataTerraformRemoteStateConsul extends TerraformRemoteState {
     super(scope, id, "consul", config);
   }
 }
-
+/**
+ * Stores the state in the Consul KV store at a given path.
+ * This backend supports state locking.
+ * 
+ * https://www.terraform.io/language/settings/backends/consul
```
Comment (ansgarm) [inline packages/cdktf/lib/backends/consul-backend.ts:43]:
```suggestion
 * Read more about this backend in the Terraform docs:
 * https://www.terraform.io/language/settings/backends/consul
```

**Review by ansgarm — COMMENTED** [review]
Nice work @aayushharwani-aidash 👏
Just two small things for parity.

**Comment by aayushharwani-aidash** [comment]
@ansgarm  made changes, as suggested by you.

File: packages/cdktf/lib/backends/azurerm-backend.ts [hunk 3]
```diff
@@ -36,21 +36,93 @@ export class DataTerraformRemoteStateAzurerm extends TerraformRemoteState {
...
+ * This backend supports state locking and consistency checking
+ * with Azure Blob Storage native capabilities.
+ * 
+ * Note: By default the Azure Backend uses ADAL for authentication which is deprecated 
+ * in favour of MSAL - MSAL can be used by setting use_microsoft_graph to true. 
+ * The default for this will change in Terraform 1.2, 
+ * so that MSAL authentication is used by default.
+ * 
+ * Read more about this backend in the Terraform docs:
+ *
```
[the remark this answers is not in the document]
Comment (ansgarm) [reply packages/cdktf/lib/backends/azurerm-backend.ts:52]: let's remove the empty line here before the link

File: packages/cdktf/lib/backends/consul-backend.ts [hunk 4]
```diff
@@ -36,18 +36,66 @@ export class DataTerraformRemoteStateConsul extends TerraformRemoteState {
     super(scope, id, "consul", config);
   }
 }
-
+/**
+ * Stores the state in the Consul KV store at a given path.
+ * This backend supports state locking.
+ * 
+ * Read more about this backend in the Terraform docs:
+ *
```
Comment (ansgarm) [inline packages/cdktf/lib/backends/consul-backend.ts:44]: let's remove the empty line here before the link, too :)

```suggestion
```

**Review by ansgarm — COMMENTED** [review]
Thanks! Let's remove the empty lines between the text and the link and we're good to go 👍

**Review by ansgarm — APPROVED** [review]

**Pull request merged**